### DISCUSSION TITLE
Building Backend-as-a-Serivce vs Data Pipeline

### DISCUSSION THREAD

**Discussion by qbig** [bacalhau-project/bacalhau#2065] [Q&A]
From the doc and intro youtube video, it seems that bacalhau is mainly built for ML and data pipeline workflows. It is possible to use it for build backend-as-a-service or serverless function platforms?

An related project that is still in drafting spec stage is https://github.com/ipvm-wg/spec

**Comment by aronchick** [comment]
You can absolutely use it as a backend-as-a-service to do arbitrary container or WASM jobs. The biggest thing will be thinking about your state and where you store it. We do not support "long running" jobs, so as long as your container can start, mount state, run the job, and then shut down, we can absolutely do it.

What's your particular use case?
